Rakefile

```diff
@@ -65,8 +65,48 @@ if saas?
     end
 
     require 'rspec/core/rake_task'
-    desc 'Run specs'
-    RSpec::Core::RakeTask.new
+    spec_task_dependencies = ['spec:unit', 'spec:integration', 'spec:acceptance', 'spec:api', 'spec:use_cases']
+
+    desc 'Run RSpec tests'
+    task :spec => spec_task_dependencies
+
+    namespace :spec do
+
+      require 'rspec/core/rake_task'
+      desc 'Run all RSpec tests (unit, integration, acceptance, api, use_cases)'
+      task :all => spec_task_dependencies
+
+      desc 'Run RSpec unit tests'
+      RSpec::Core::RakeTask.new(:unit) do |task|
+        task.pattern = 'spec/unit/**/*_spec.rb'
+      end
+
+      desc 'Run RSpec integration tests'
+      RSpec::Core::RakeTask.new(:integration) do |task|
+        task.pattern = 'spec/integration/**/*_spec.rb'
+      end
+
+      desc 'Run RSpec acceptance tests'
+      RSpec::Core::RakeTask.new(:acceptance) do |task|
+        task.pattern = 'spec/acceptance/**/*_spec.rb'
+      end
+
+      desc 'Run RSpec api tests'
+      RSpec::Core::RakeTask.new(:api) do |task|
+        task.pattern = 'spec/api/**/*_spec.rb'
+      end
+
+      desc 'Run RSpec use_cases tests'
+      RSpec::Core::RakeTask.new(:use_cases) do |task|
+        task.pattern = 'spec/use_cases/**/*_spec.rb'
+      end
```

Comment:
Nice job - however, when I look at all these rules there is a DRY flag being raised: they all do the same except for a very small thing. This would be a bit of advanced stuff, but can we come up with a way to have Ruby write all these for us?